### ISSUE TITLE
tf.keras.callbacks.ReduceLROnPlateau -  min_delta parameter should be percentage, not absolute  

### ISSUE THREAD

**Issue by danieltomer** [tensorflow/tensorflow#33675]
**System information**
- TensorFlow version 2.0:

**Describe the feature and the current behavior/state.**
Regarding tf.keras.callbacks.ReduceLROnPlateau: The min_delta parameter is currently an absolute number which indicates when a meaningful reduction in the monitored loss has accrued. It makes no sense to use an absolute number for two reasons - 
1. Every loss has a different dynamic range and hence a different definition for a meaningful reduction
2. A "meaningful reduction" decreases as the training progresses. The higher the epoch the smaller of a change in loss is expected.  

For these two reasons I think that a percentage of change in the monitored loss is much more useful. 

**Will this change the current api? How?**
Maybe, if the implantation is by adding a new parameter to the callback named "min_delta_percent".
**Who will benefit with this feature?**
Everyone using this callback.

**Comment by ravikyram** [comment]
Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

**Comment by danieltomer** [comment]
The case is whenever this callback is used. It always makes more sense to use a min_delta_percent than an absolute min delta, As explained above

**Comment by ruancomelli** [comment]
Any updates on this? I am interested in this functionality as well for the same reasons as listed above, especially reason 2.

In the problem I'm trying to solve, the loss is ~1e5 at the beginning of training, while the goal is to achieve a loss as low as 10 at the end. One can easily see that `min_delta=10` has very different meanings in the beginning and in the end.

As a workaround, I copy-pasted Tensorflow's `ReduceLROnPlateau` into my project and added an initialization parameter called `min_delta_mode`, defaulting to `"absolute"`. In this case, the callback works as it does presently, with
```py
self.monitor_op = lambda current, best: np.less(current, best - self.min_delta)
```
However, when `min_delta_mode == "relative"`, I define 
```py
self.monitor_op = lambda current, best: np.less(current, (1 - self.min_delta)*best)
```
It works as expected.

**Comment by jvishnuvardhan** [comment]
@danieltomer @ruancomelli Sorry for the late response.  Did you test the feature for different models? Did you notice consistently better results with this feature?

Are you still interested in contributing? If yes, please feel free to open a PR in  [keras-team/keras repo.](https://github.com/keras-team/keras/issues) repository.

Please note that Keras development moved to keras-team/keras repository to focus on only keras. Thanks!

**Comment by ruancomelli** [comment]
@jvishnuvardhan I tested it only with the models I was studying, which are simple convolutional networks like:
```py
x = Input(shape=input_shape)
x = LayerNormalization()(x)
x = Conv2D(
    32,
    (5, 5),
    padding='same',
    activation='relu',
    dtype=hidden_layers_policy,
)(x)
x = MaxPool2D((2, 2), strides=(2, 2), dtype=hidden_layers_policy)(x)
x = Dropout(dropout, dtype=hidden_layers_policy)(x)
x = Flatten(dtype=hidden_layers_policy)(x)
x = Dense(200, activation='relu', dtype=hidden_layers_policy)(x)
x = Dropout(dropout, dtype=hidden_layers_policy)(x)
x = Dense(1, dtype=hidden_layers_policy)(x)
predictions = Activation('linear', dtype=output_layer_policy)(x)

model = Model(inputs=input_data, outputs=predictions)
```
and some variations adding more `Conv2D` or `Dense` layers as well as changing those layers' sizes.

I noticed that using `min_delta_mode="relative"` consistently improved my models' performances because of the reasons mentioned before, but I haven't tested this exhaustively with other kinds of models or datasets.

I would really love to contribute to TensorFlow/Keras, yes! Do you know which kinds of tests would be required to justify such a feature, or is it enough to submit a PR containing the changes suggested before?

**Comment by jvishnuvardhan** [comment]
@ruancomelli Please submit a PR in keras-team/keras repo. Keras-team will check it and ask you to add a unit test. Thanks for your contribution. Thanks!
